presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -403,7 +403,12 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                 leftSource = new ProjectNode(idAllocator.getNextId(), leftSource, leftProjections.build());
                 rightSource = new ProjectNode(idAllocator.getNextId(), rightSource, rightProjections.build());
 
-                output = new JoinNode(node.getId(), node.getType(), leftSource, rightSource, joinConditionBuilder.build(), newJoinFilter, node.getLeftHashSymbol(), node.getRightHashSymbol());
+                List<Symbol> outputSymbols = ImmutableList.<Symbol>builder()
```

Comment:
It might produce more outputs indeed, but `PruneUnreferencedOutputs` should remove those new unused symbols. I wanted to separate concerns in this case to make make predicate pushdown cleaner. 

Anyway, I have applied your comment in additional commit: https://github.com/prestodb/presto/pull/6501/commits/7a2aaaf130ae0e238280bf51fdc4554972f5dc8e since it is easy to apply

Anyway,